proposals/p4105.md

```diff
@@ -0,0 +1,305 @@
+usable in their environment. Carbon should again embrace this need and see the
+standardization as an analogous process to promoting a normal release into an
+LTS. Some relevant and effective LTS should be selected and taken through
+whatever process is identified to create a standard reflecting that LTS version
+of Carbon. Updates to the standard should in turn track as updates to a newer
+LTS. The specifics of how to do this are left to the future work, and they may
+change exactly how this works.
+
+Note that the goal of this future direction isn't to constrain how Carbon can
+arrive effectively at either an LTS release or a standard. Instead, the goal is
+to make it clear that we _should_ be open and planning to achieve these in order
+to meet the needs of candidate Carbon users.
+
+## Rationale
+
+-   [Language tools and ecosystem](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#language-tools-and-ecosystem)
+    -   Carbon needs a coherent versioning scheme across the language itself as
+        well as its ecosystem. Especially as the language is developing rapidly,
+        being able to sync across all of these with a single, simple versioning
+        scheme is especially important to have the tooling and ecosystem agree
+        about features of the language.
+-   [Software and language evolution](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#software-and-language-evolution)
+    -   Users of Carbon need a clear model for understanding the language's
+        evolution, tracking it, and responding to significant aspects of it.
+    -   The versioning scheme needs to support a wide variety of versions built
+        in the process of evolving the language without creating confusion.
+
+## Alternatives considered
+
+### Do nothing, or just talk about a minimal nightly version.
+
+Advantages:
+
+-   Carbon is at a very early stage and is a long way from needing release
+    candidates for stable releases or incrementing its major version numbers.
+-   No need to imagine distant future scenarios.
+
+Disadvantages:
+
+-   We already discuss version numbers in varyous places in the Carbon project,
+    including our [milestones](/docs/project/milestones.md) and
+    [roadmap](/docs/project/roadmap.md).
+-   Establishing how the Carbon project will communicate its updates with
+    version number changes in advance of those updates makes that communication
+    more effective.
+-   Lets us telegraph our intentions and how we are thinking about releasing and
+    cadence to the community and get earlier feedback.
+
+A key to adopting the more detailed versioning plan is that we can change any
+and all of this if and when we need to. This does not lock Carbon into using
+this exact versioning scheme. We should actively collect and listen to feedback
+in this space from potential users and customers.
```

Comment:
Certainly we should listen to feedback, but is it worth *actively* collecting it (eg, going out and soliciting information)? I don't think it's worth (for example) surveying our current and prospective users to find what they think about our versioning scheme -- if no-one proactively gives feedback I think we can assume it's working well enough.

Reply:
Agreed, done.